nixos/modules/services/web-apps/libretranslate.nix

```diff
@@ -0,0 +1,203 @@
+        type = lib.types.listOf lib.types.str;
+        default = [ ];
+        example = [
+          "--debug"
+          "--disable-files-translation"
+        ];
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    environment.systemPackages = [ ltmanageKeysCli ];
+
+    systemd.tmpfiles.rules = lib.mkIf (cfg.dataDir != "/var/lib/libretranslate") [
+      "d '${cfg.dataDir}' 0750 ${cfg.user} ${cfg.group} - -"
+      "z '${cfg.dataDir}' 0750 ${cfg.user} ${cfg.group} - -"
+    ];
+
+    systemd.services.libretranslate = {
+      description = "LibreTranslate service";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      environment = {
+        HOME = cfg.dataDir;
+      };
+      serviceConfig = lib.mkMerge [{
+        Type = "simple";
+        ExecStart = ''
+          ${cfg.package}/bin/libretranslate \
+            --host ${cfg.host} \
+            --port ${toString cfg.port} \
+            ${lib.optionalString cfg.enableApiKeys "--api-keys"} \
+            ${lib.optionalString cfg.disableWebUI "--disable-web-ui"} \
+            ${lib.optionalString cfg.updateModels "--update-models"} \
+            ${lib.optionalString (cfg.threads != null) "--threads ${toString cfg.threads}"} \
+            ${lib.concatStringsSep " " cfg.extraArgs}
+        '';
+        WorkingDirectory = cfg.dataDir;
+        # User and group
+        User = cfg.user;
+        Group = cfg.group;
+        # Proc filesystem
+        ProcSubset = "all";
+        ProtectProc = "invisible";
+        # Access write directories
+        UMask = "0027";
+        # Capabilities
+        CapabilityBoundingSet = "";
+        # Security
+        NoNewPrivileges = true;
+        # Sandboxing
+        ProtectSystem = "strict";
+        ProtectHome = true;
+        PrivateTmp = true;
+        PrivateDevices = true;
+        PrivateUsers = true;
+        ProtectHostname = true;
+        ProtectClock = true;
+        ProtectKernelTunables = true;
+        ProtectKernelModules = true;
+        ProtectKernelLogs = true;
+        ProtectControlGroups = true;
+        RestrictAddressFamilies = [ "AF_INET" "AF_INET6" ];
+        RestrictNamespaces = true;
+        LockPersonality = true;
+        MemoryDenyWriteExecute = false;
+        RestrictRealtime = true;
+        RestrictSUIDSGID = true;
+        RemoveIPC = true;
+        PrivateMounts = true;
+        # System Call Filtering
+        SystemCallArchitectures = "native";
+        SystemCallFilter = [ "~@cpu-emulation @debug @ipc @keyring @mount @obsolete @privileged @setuid" ];
+      }
+      (lib.mkIf (cfg.dataDir == "/var/lib/libretranslate") {
+        # State directory and mode
+        StateDirectory = "libretranslate";
+        StateDirectoryMode = "0750";
+      })
+      (lib.mkIf (cfg.dataDir != "/var/lib/libretranslate") {
+        ReadWritePaths = cfg.dataDir;
+      })];
+    };
+
+    services.nginx = lib.mkIf cfg.configureNginx {
```

Comment:
I think this is out of scope of this module. The moment someone will want to configure, for example, https, or maybe proxy_cache, it easier to just do it manually.

Reply:
This is not out of scope at all. A lot of bigger modules like nextcloud do the same to make it easy to use the service.

> for example, https, or maybe proxy_cache

It isn't, you can just configure https for the virtualHost and the domain and the module system will merge the settings together.